pycmr/granule.py

```diff
@@ -152,3 +156,69 @@ def execute(self):
 
         response = get(url)
         return response.json()
+
+    def downloadable(self, downloadable):
+        """
+        Set the downloadable value for the query.
+
+        Must be of type Boolean
+        """
+        if not isinstance(downloadable, bool):
+            raise TypeError("Downloadable must be of type bool")
+
+        self.params['downloadable'] = downloadable
+
+        return self
+
+    def online_only(self, online_only):
+        """
+        Set the online_only value for the query.
+
+        Must be of type Boolean
+        """
+
+        if not isinstance(online_only, bool):
+            raise TypeError("Online_only must be of type bool")
+
+        self.params['online_only'] = online_only
+
+        return self
+
+    def entry_title(self, entry_title):
+        """
+        Set the entry_title value for the query
+        """
+
+        entry_title = self._urlEncodeString(entry_title)
+
+        self.params['entry_title'] = entry_title
+
+        return self
+
+    def orbit_number(self, orbit_number):
+        """"
+        Set the orbit_number value for the query
+        """
+
+        try:
+            self.params['orbit_number'] = int(orbit_number)
+        except ValueError:
+            self.params['orbit_number'] = self._urlEncodeString(orbit_number)
```

Comment:
Ah yeah, that would be. What you'd want in that case is to simply re-raise the ValueError exception, but perhaps with a new message like "Please provide orbit_number as an integer" just to nudge people in the right direction. That way it will still handle int/float/string if they cast correctly, and otherwise raises a useful exception to the caller.